pkg/kapis/oauth/handler.go

```diff
@@ -212,23 +204,59 @@ func (h *oauthHandler) OAuthCallBackHandler(req *restful.Request, resp *restful.
 		return
 	}
 
+	result, err := h.issueTo(user.GetName())
+	if err != nil {
+		err := apierrors.NewUnauthorized(fmt.Sprintf("Unauthorized: %s", err))
+		resp.WriteError(http.StatusUnauthorized, err)
+		return
+	}
+	h.im.AuthSuccessRecord(user.GetName(), result.AccessToken)
+	resp.WriteEntity(result)
+}
+
+func (h *oauthHandler) Login(request *restful.Request, response *restful.Response) {
+	var loginRequest auth.LoginRequest
+
+	err := request.ReadEntity(&loginRequest)
+	if err != nil || loginRequest.Username == "" || loginRequest.Password == "" {
+		response.WriteHeaderAndEntity(http.StatusUnauthorized, fmt.Errorf("incorrect username or password"))
```

Comment:
Shouldn't the error message be "empty username or password" ?

Reply:
fixed